compiler/AST/FnSymbol.cpp

```diff
@@ -118,6 +120,19 @@ void FnSymbol::verify() {
     INT_FATAL(this, "Bad FnSymbol::lifetimeConstraints::parentSymbol");
   }
 
+  // constrainedTypes: AList of DefExpr of ConstrainedType
+  INT_ASSERT(constrainedTypes.parent == this);
+  for_alist(ctExpr, constrainedTypes) {
+    Symbol* ctSym = toDefExpr(ctExpr)->sym;
```

Comment:
maybe assert isDefExpr and then assert isTypeSymbol so we get internal errors rather than core dump if the verify fails?